sdk/go/common/workspace/project.go

```diff
@@ -143,8 +156,92 @@ type Project struct {
 	AdditionalKeys map[string]interface{} `yaml:",inline"`
 }
 
-func ValidateProject(raw interface{}) error {
-	// Cast any map[interface{}] from the yaml decoder to map[string]
+func isPrimitiveValue(value interface{}) (string, bool) {
```

Comment:
We should include `float64` here.

Playing code gold, I think this would be better expressed as:
```go
switch value.(type) {
case bool:
    return "boolean", true
case int:
    return "integer", true
...
}
```

Reply:
+1 to Ians suggestion to rewrite this as a switch